pkgs/applications/science/math/wolfram/default.nix

```diff
@@ -59,13 +59,22 @@ let versions = callPackage ./versions.nix { };
 in
 
 callPackage ./generic.nix {
+
   inherit cudaSupport cudaPackages;
+
   inherit (found-version) version lang;
+
   src = if source == null then found-version.src else source;
-  name = ("mathematica"
-          + lib.optionalString cudaSupport "-cuda"
-          + "-${found-version.version}"
-          + lib.optionalString (lang != "en") "-${lang}");
+
+  name = let
+    basename = if lib.versionAtLeast found-version.version "14.1.0"
+               then "wolfram"
+               else "mathematica";
+  in (basename
+      + lib.optionalString cudaSupport "-cuda"
+      + "-${found-version.version}"
+      + lib.optionalString (lang != "en") "-${lang}");
+
   meta = with lib; {
     description = "Wolfram Mathematica computational software system";
```

Comment:
Maybe , it'd be nice to mention something in the `description` for the new version? Or in `meta.longDescription`?

Reply:
There are potentially more things to clear up - but only after we have figured out the direction.